docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -862,6 +863,50 @@ We can include the tag like this:
 <1> `item` is resolved to an iteration element and can be referenced using the `it` key in the tag template.
 <2> Tag content injected using the `nested-content` key in the tag template.
 
+=== Rendering Output
+
+`TemplateInstance` provides several ways to trigger the rendering and consume the result.
+The most straightforward approach is represented by `TemplateInstance.render()`.
+This method triggers a synchronous rendering, i.e. the current thread is blocked until the rendering is finished, and returns the output.
+By contrast, `TemplateInstance.renderAsync()` returns a `CompletionStage<String>` which is completed when the rendering is finished.
+
+.`TemplateInstance.renderAsync()` Example
+[source,java]
+----
+template.data(foo).renderAsync().whenComplete((result, failure) -> { <1>
+   if (failure == null) {
+      // consume the output...
+   } else {
+      // process failure...
+   }
+};
+----
+<1> Register a callback that is executed once the rendering is finished. 
+
+There are also two methods that return https://smallrye.io/smallrye-mutiny/[Mutiny] types.
+`TemplateInstance.createUni()` returns a new `Uni<String>` object.
```

Comment:
What about `asUni`? and `asMulti`?

Reply:
Hm, I don't know. I usually only use `asX()` methods to easily cast a more generic type to a subtype, e.g. something like `org.jboss.jandex.AnnotationTarget.asMethod()`.

Reply:
as you have `renderAsync`, maybe `renderAsUni` and `renderAsMulti`?

Reply:
The thing is that `createUni()` and `createMulti()` don't trigger rendering. It's the subscription that does trigger it.

Reply:
Right! 

So, let's keep `createUni` and `createMulti`.